### ISSUE TITLE
[Bug]: QuadMesh performance issues on mouse-move

### ISSUE THREAD

**Issue by raphaelquast** [matplotlib/matplotlib#23164]
### Bug summary

Hey  
I've noticed that there are quite some performance-issues with `collections.QuadMesh` in [EOmaps](https://github.com/raphaelquast/EOmaps) for large datasets and so I had a bit closer look on what's going on...

From what I see, the main issue originates from the fact that `QuadMesh.contains()` is called twice on each mouse-movement:

https://github.com/matplotlib/matplotlib/blob/4d7c309190745daa861102ec9c6c72766cc1047f/lib/matplotlib/backend_bases.py#L3000-L3001

https://github.com/matplotlib/matplotlib/blob/4d7c309190745daa861102ec9c6c72766cc1047f/lib/matplotlib/collections.py#L2186-L2187

Now, I guess the problem here is that `QuadMesh.contains()` acts differently from other collections... to quote from the docs:

    Unlike other `.Collection`, the default *pickradius* of `.QuadMesh` is 0,
    i.e. `~.Artist.contains` checks whether the test point is within any of the
    mesh quadrilaterals.

This contains-test is particularly slow for large datasets (which I guess is the primary use for QuadMeshes in the first place) 
and consequently executing it on each mouse-move-event introduces a huge lag in the response-time of the plot dependent on the size of the dataset. 

(this is again somewhat connected to #22334, but not really...)

### Code for reproduction
just compare the performance of this:
```python
import matplotlib.pyplot as plt
import numpy as np

np.random.seed(19680801)
Z = np.random.rand(1000, 1000)
x = np.linspace(0, 10, 1000) 
y = np.linspace(10, 20, 1000) 

fig, ax = plt.subplots()
qmesh = ax.pcolormesh(x, y, Z)
```
 ... with the performance of this:
```python
import matplotlib.pyplot as plt
import numpy as np

np.random.seed(19680801)
Z = np.random.rand(1000, 1000)
x = np.linspace(0, 10, 1000) 
y = np.linspace(10, 20, 1000) 

fig, ax = plt.subplots()
qmesh = ax.pcolormesh(x, y, Z)
qmesh.contains = lambda *args, **kwargs: [False]
```

### Actual outcome

figure is very laggy

### Expected outcome

figure response-time is as nice as usual 🙂 

### Operating system

Windows

### Matplotlib Version

3.5.1

### Matplotlib Backend

Qt5Agg

### Python version

3.7.12

### Installation

conda

**Comment by jklymak** [comment]
This was fixed in 3.5.2, right?

**Comment by raphaelquast** [comment]
oh... sorry seems like I missed that release 😅 
yep things look a lot better in 3.5.2   ... thanks!
